lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -526,40 +526,111 @@ LogicalResult ConvertAtenOp<AtenArgmaxOp>::matchAndRewrite(
   return success();
 }
 
-template <>
-LogicalResult ConvertAtenOp<AtenSqueezeOp>::matchAndRewrite(
-    AtenSqueezeOp op, OpAdaptor adaptor,
-    ConversionPatternRewriter &rewriter) const {
+template <typename AtenOpT>
+class ConvertAtenSqueezeOp : public OpConversionPattern<AtenOpT> {
+public:
+  using OpConversionPattern<AtenOpT>::OpConversionPattern;
+  using OpAdaptor = typename AtenOpT::Adaptor;
 
-  Value self = adaptor.self();
-  auto selfTy = self.getType().template cast<RankedTensorType>();
+  // Each variant must implement corresponding parameter parsing options
+  virtual LogicalResult
+  generateSqueezedShape(AtenOpT op, RankedTensorType selfTy,
+                        ConversionPatternRewriter &rewriter,
+                        SmallVector<int64_t> &squeezedShape) const {
+    return rewriter.notifyMatchFailure(
+        op, "Unimplemented dim/dim-list parsing function");
+  }
 
-  if (!selfTy)
-    return op.emitError("Only ranked tensor types supported in TOSA argmax");
+  // Common rewriter for all squeeze ops, calls the specific implementation of
+  // generateSqueezedShape() needed for the op variant.
+  LogicalResult matchAndRewrite(AtenOpT op, OpAdaptor adaptor,
+                                ConversionPatternRewriter &rewriter) const {
```

Comment:
Many of the `matchAndRewrite` functions in this file (including this one) are generating compiler warnings because they're not marked as `override`